numba/core/utils.py

```diff
@@ -188,6 +188,18 @@ class ConfigStack:
     """
     tls = threading.local()
 
+    @classmethod
+    def top_or_none(cls):
+        """Get the TOS of return None if no config is set.
+        """
+        self = cls()
+        if self:
+            flags = self.top()
+        else:
+            # Note: should this be the default flag for the target instead?
```

Comment:
I think perhaps it should be the default so it is always known precisely which flags are in use for a given function.

Reply:
I'm getting into a rabbit hole trying to put a default flag there. Currently, the `TargetOptions` is not really part of any contexts. It's floating in the `TargetDescriptor`. I can put the options into the target-context but it doesn't make sense for it to appear in the typing-context. However, I need it during typing. This goes back to the purity problem of needing flags (which I don't think is part of types) during the typing phase.

The only doable hack I can think of is putting the info into the `ConfigStack` to read the default flag later.

Reply:
hm.... the other way is to let flags be part of the typing context.

Reply:
I sampled the testsuite and found that this branch is only reached from test that bypass the user-facing compiler API---e.g. tests that invokes part of the compiler to test for specific internal behavior. The failure to use a default flag will only result in redundant compilation. I don't think we should spend too much time on this problem for now.

Reply:
From OOB conversation, agree on leaving it for now as this fixes the immediate issue.